dotnet/src/webdriver/Remote/RemoteSessionSettings.cs

```diff
@@ -306,34 +301,42 @@ private List<object> GetFirstMatchOptionsAsSerializableList()
             return optionsMatches;
         }
 
-        private bool IsJsonSerializable(object arg)
+        private static bool IsJsonSerializable(object arg)
```

Comment:
What if we remove this method at all? Because this class doesn't know about serialization context? It cannot determine whether something can be serialized..

Reply:
This logic was introduced when the `RemoveSessionSettings` was created 6 years ago 0b310c111486897cc7c711855ee9dc9005fc2dc3

The original explanation was `The value to be set must be serializable to JSON for transmission across the wire to the remote end.`. However, models are perfectly JSON serializable. I don't see any potential problem with this.

On the other hand, it makes the serialization very confined, since we know all possible types... Especially since the `ToString()` implementation is JSON serializing the instance (not sure if anyone relies on this ToString behavior, hopefully not).

I think if we change the `ToString()` behavior to behave like a normal, cheap `ToString()` instead of really serializing the type, then it's safe to remove the restriction.

Reply:
Yes, let's do it together. While we are here in context of json serialization, then let's "break" it. We already broke it, you know.

If we are going to normalize a usage of `ToString()` methods, then we should have a suggestion for users who rely on this changed behavior.

Reply:
PR has been updated to allow any type to be JSON-serialized, not just the "primitives", like a reverse `ResponseValueJsonConverter`.

Reply:
Road to simplicity, thank you!